filebeat/tests/system/test_modules.py

```diff
@@ -107,7 +107,7 @@ def load_fileset_test_cases():
                 continue
 
             test_files = glob.glob(os.path.join(modules_dir, module,
-                                                fileset, "test", "*.log"))
+                                                fileset, "test", os.getenv("TESTING_FILEBEAT_FILEPATTERN", "*.log")))
```

Comment:
@ph @cmacknz General problem we have with these environment variables is that everyone that worked on the test suite knows about these, everyone else has to find them in the code base ... We should find a better place (not related to this PR).

Reply:
Agree

I think we should rethink how and why we introduce variables in the first place. The problem today is there are multiple ways to run tests that support or require different side effects.

I strongly think the test cli should give me all the required information for options without looking in the documentation.

Reply:
Definitely agreed.

Not sure if pytest offers something like this directly. `mage pythonIntegTest` has some help, but does a lot of extra work (15m).

I guess maybe one option would be a lighter mage target with these options covered in the inline help.

In working on the tests I also noticed some similarity with metricbeat `testdata` integration tests https://github.com/elastic/beats/blob/e6839abb1bab5a7ff89ebbf6940703ba401696a0/metricbeat/module/couchbase/bucket/bucket_test.go#L36

If we did these filebeat module tests as regular golang test cases, stuff like `TESTING_FILEBEAT_FILEPATTERN` would just be `go test -run (pattern)`

Reply:
I feel that. 🧡